From 5cface3df0e0213d8bc593d82a9a7c1e648dd71a Mon Sep 17 00:00:00 2001
From: Mahendra Narvariya <mahendra84@users.sourceforge.net>
Date: Thu, 19 Nov 2015 13:42:30 +0100
Subject: [PATCH 4/8] Patch to remove redundant comparison in file mmsh.c

Variable uri can not be NULL at line no 665 as NULL check is already
applied at line no 625.
---
 src/mmsh.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/mmsh.c b/src/mmsh.c
index a019f05..dc17c9d 100755
--- a/src/mmsh.c
+++ b/src/mmsh.c
@@ -662,10 +662,10 @@ mmsh_t *mmsh_connect (mms_io_t *io, void *data, const char *url, int bandwidth)
     gnet_uri_delete(proxy_uri);
     proxy_uri = NULL;
   }
-  if (uri) {
-    gnet_uri_delete(uri);
-    uri = NULL;
-  }
+
+  gnet_uri_delete(uri);
+  uri = NULL;
+
   if (!mms_valid_proto(this->proto)) {
     lprintf("unsupported protocol\n");
     goto fail;
-- 
2.39.2

